File: Challenges/Leetcode/49 (Medium)/Python/Solution.py
"""
  Breeland Abrams
  Leetcode 49: Group Anagrams
""""
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        count = [0] * 26
        m = {}
        
        for i in range(len(strs)):
            for j in range(len(strs[i])):
                count[ord(strs[i][j]) - 97] += 1
            if(m.get("".join(str(count)), None) is not None):
                m["".join(str(count))].append(strs[i])
            else:
                m["".join(str(count))] = [strs[i]]
            count = [0]*26
            
        return [m[_] for _ in m.keys()]
        

File: Challenges/Leetcode/49 (Medium)/Java/Solution.java

/*
  Breeland Abrams
  Leetcode 49: Group Anagrams
*/

import java.util.ArrayList;
import java.util.HashMap;

class Solution {
  public List<List<String>> groupAnagrams(String[] strs) {

    int[] count = new int[26];
    HashMap<String, ArrayList<String>> map = new HashMap<String, ArrayList<String>>();

    for (int i = 0; i < strs.length; i++) {
      Arrays.fill(count, 0);
      for (char c : strs[i].toCharArray()) {
        count[c - 97] += 1;
      }

      if (map.get(Arrays.toString(count)) != null) {
        map.get(Arrays.toString(count)).add(strs[i]);
        continue;
      }
      map.put(Arrays.toString(count), new ArrayList<String>(Arrays.asList(strs[i])));

    }

    return new ArrayList<>(map.values());
  }
}